src/app/get-involved/get-involved.component.ts

```diff
@@ -10,16 +12,21 @@ import { GEOCHART_CONFIG } from './chart-configuration.model';
 })
 export class GetInvolvedComponent implements OnInit {
 
+  isLoading = true;
   data;
   chart;
 
-  constructor(private getInvolvedService: GetInvolvedService) { }
+  constructor(private countrySupportersService: CountrySupportersService) { }
 
   ngOnInit() {
-    this.getInvolvedService.getInvolvedPeopleByCountry()
-      .subscribe((involvedPeopleByCountry) => {
-        this.data = involvedPeopleByCountry
-          .map(([countryName, countryCount]) => [countryName, countryCount, this.generateHtmlTooltip(countryName, countryCount)]);
+    this.countrySupportersService.getAll()
+      .pipe(
+        map((countrySupporters) =>
+          countrySupporters.map(({count, country}) => [country, count, this.generateHtmlTooltip(country, count)])
+      ))
+      .subscribe((supporters) => {
+        this.isLoading = false;
+        this.data = supporters;
         this.chart = Object.assign({ data: this.data }, GEOCHART_CONFIG);
```

Comment:
You should unsubscribe from this when the component is destroyed. I don't think we want to continue doing this if the component is not being used. Maybe not here, but when the app grows, we could ran into random and difficult to identify problems for not killing the subscription of an observable of Firebase.

Reply:
uhh good catch! :D yeap, i will add the unsubscribe, but i think as we are using lazy loading if the component is not visible basically all the service under that module will be destroyed. But having it in case we change the lazy loading approach won't hurt.

Reply:
Done